### ISSUE TITLE
Handling variable length reads and pre-trimmed primers

### ISSUE THREAD

**Issue by michael-weinstein** [Zymo-Research/figaro#17]
Due to seeing more pipeline changes that result in a need for handling variable length reads in scenarios where I do not expect the trimming to invalidate the model, I am working on making an update to the program that will allow for this.  Likewise, I am going to enable handling primers of length zero for when primers are pretrimmed.

**Comment by mikemc** [comment]
@michael-weinstein Just checking if the case zero-length primers will be fixed any time soon, as I'm working with data from a protocol where the primer region is not sequenced. There are several common amplicon protocols like this that don't sequence the primer region at all, so the inability to set the primer length to 0 is going to be a deal breaker for many people.

**Comment by mikemc** [comment]
In the meantime, is there anything wrong with the following hack work, for the case where the reads lack primer sequences? Set the primer lengths to 1 (or whatever the smallest number is that doesn't cause an error) and decrease the requested overlap by 1.

**Comment by michael-weinstein** [comment]
Yes, I think that should work for this.  I am working on an update to cover that, but am currently saturated with COVID-related requests.

**Comment by gregdenay** [comment]
With reads that are pre-processed to remove primers and be trimmed to a given length (I do it with cutadapt but i guess there are other possibilities), I can confirm that the above answer works.

**Comment by padpadpadpad** [comment]
Hi @michael-weinstein just wondering if there is any progress on Figaro being able to handle variable read lengths. I have an amplicon project where primers are already trimmed and I have read lengths of 248/249/250 bp. 

Get an error `Forward read files appear to be of different lengths or of varied lengths.` I am using the python command line version at the moment.

Many thanks
Dan

**Comment by janetw** [comment]
Hello, I am wondering if there is any progress with Figaro being able to handle reads already trimmed of their primers? In the meantime I will try the above suggestion. Thanks!

**Comment by michael-weinstein** [comment]
I have been working on this on and off recently, but I have been tied up working on a few COVID-related apps (Mara, Virsieve, and ZymoTransmit) and have needed to prioritize those for the last several months.  It's still very much in my plans to make that update.

**Comment by janetw** [comment]
Thank you for the update. I just posted in another about the varied lengths and then came back to this post to reply to you. I was initially focused on having pretrimmed reads and wasn't thinking about the varied lengths at the time. So I guess if we have varied lengths, it doesn't work for me at this time?

**Comment by bernt-matthias** [comment]
Seems that  https://github.com/Zymo-Research/figaro/pull/43 could be a solution? But nobody of the maintainers cared to respond. So I'm wondering if this project is still alive?
